flutter_app/lib/widgets/waypoint_queue_widget.dart

```diff
@@ -0,0 +1,186 @@
+  final TextEditingController _altitudeInput = TextEditingController();
+
+  late double _latitude;
+  late double _longitude;
+  late double _altitude;
+
+  /// Gets the waypoints from the TextFields
+  void _getWaypointsFromInput() {
+    try {
+      _latitude = double.parse(_latitudeInput.text);
+      _longitude = double.parse(_longitudeInput.text);
+      _altitude = double.parse(_altitudeInput.text);
+    } catch (e) {
+      print('Enter valid numbers.');
+    }
+  }
+
+  /// Sends a command to queue a waypoint.
+  void _queueWaypoint() {
+    _getWaypointsFromInput();
+    widget.queueWaypoints.queueWaypoint(
+      widget.systemId,
+      widget.componentId,
+      _latitude,
+      _longitude,
+      _altitude,
+    );
+    setState(() {});
+  }
+
+  /// Sends a command to send a waypoint, bypassing the queue.
+  void _queueWaypointWithoutQueue() {
+    _getWaypointsFromInput();
+    widget.queueWaypoints.sendWaypointWithoutQueue(
+      widget.systemId,
+      widget.componentId,
+      _latitude,
+      _longitude,
+      _altitude,
+    );
+  }
+
+  /// Sends the first waypoint in the queue to the drone.
+  void _sendNextWaypointInQueue() {
+    widget.queueWaypoints.sendNextWaypointInQueue();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const Text(
+          'Waypoint Queue',
+          style: TextStyle(fontWeight: FontWeight.bold),
+        ),
+        Expanded(
+          child: ListView(
+            children: [
+              DataTable(
+                columns: const [
+                  DataColumn(label: Text('Latitude')),
+                  DataColumn(label: Text('Longitude')),
+                  DataColumn(label: Text('Altitude')),
+                ],
+                rows: widget.queueWaypoints.waypointQueue
+                    .map(
+                      ((element) => DataRow(
+                            cells: <DataCell>[
+                              DataCell(Text(element.x.toString())),
+                              DataCell(Text(element.y.toString())),
+                              DataCell(Text(element.z.toString())),
+                            ],
+                          )),
+                    )
+                    .toList(),
+              ),
+              const SizedBox(height: 16),
+              ElevatedButton(
+                  onPressed: _sendNextWaypointInQueue,
+                  child: const Text('Send Next Waypoint in Queue to Drone')),
+              const SizedBox(height: 16),
+              const Text(
+                'Enter a Waypoint Below',
+                style: TextStyle(fontWeight: FontWeight.bold),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(10.0),
+                child: Row(
+                  mainAxisSize: MainAxisSize.min,
+                  children: [
+                    SizedBox(
```

Comment:
Instead of using `SizedBox` for the `TextFields`, try to make them dynamically fill the size they have. Only start implementing this change once you wrap everything in a `SizedBox` with fixed width though, otherwise it won't render.